### PR TITLE
Import galacticus subhalos example

### PR DESCRIPTION
**Pull request by cgannonucm**
Implements the following changes
- Added example_notebooks/from_galacticus.ipynb
- Added an id property to track galacticus subhalos

### CODE REVIEW

**Comment by dangilman** [comment]
@cgannonucm could you please add test functions for the classes and methods you created? You can have a look at the test suites in pyHalo located in pyhalo/tests

**Comment by dangilman** [comment]
@cgannonucm I think the tests are failing because the syntax you used to merge dictionaries isn't supported in python 3.8

File: pyHalo/PresetModels/external.py
Comment (dangilman) [inline pyHalo/PresetModels/external.py:83]: I assume tree_index should be an integer? I would add an else clause here that reads 

else: 
    tree_index = int(np.round(tree_index))
    
so that you have the option of sampling tree_index from a distribution

File: pyHalo/PresetModels/external.py
Comment (dangilman) [inline pyHalo/PresetModels/external.py:96]: I would change how proj_plane_normal is specified in the function call so that the orientation of the subhalo population can be randomly sampled. The sampling done in quadmodel is for individual parameters, so you'll need to break up proj_plane_normal into its components. 

I suggest adding three arguments to the function with default values projection_x=1, projection_y=0, projection_z=0. You then have the option of sampling values of projection_x, projection_y, and projection_z from a prior to generate random orientations.

File: pyHalo/PresetModels/external.py
Comment (dangilman) [inline pyHalo/PresetModels/external.py:89]: Instead of using the CDM preset model and kwargs_cdm, can you instead pass in the name of a preset model with (that defaults to CDM) preset_model_los='CDM', load the preset model using the preset_model_from_name function, and then rename "kwargs_cdm" to "kwargs_LOS"? This would allow us to combine galacticus with any existing pyHalo model for line-of-sight halos

**Review by dangilman — CHANGES_REQUESTED** [review]

File: pyHalo/PresetModels/external.py
Comment (cgannonucm) [reply pyHalo/PresetModels/external.py]: Done. I also changed the function to only accept angle arguments instead of angle arguments and vector arguments. This way the inputs are simpler, and passing in an angle is probably more natural when sampling from a distribution.

**Comment by dangilman** [comment]
@cgannonucm ok everything looks great! Do you need to make any changes to the example notebook so that it is compatible with the latest changes? If not, then I'll go ahead and merge this

**Comment by cgannonucm** [comment]
The notebooks should be good to go! I would like to push one more commit, before we merge. I am going to add the parameter file used to generate the output file.

**Comment by cgannonucm** [comment]
@dangilman Everything should be good to go after tests finish up.

**Pull request merged**
